src/app/containers/Caption/index.jsx

```diff
@@ -9,22 +9,32 @@ import InlineLink from '../InlineLink';
 
 const componentsToRender = { fragment: Fragment, urlLink: InlineLink };
 
-const renderText = block => {
-  const textBlocks = block.model.blocks[0].model.blocks[0].model.blocks;
+const renderText = textBlocks => (
+  <Blocks blocks={textBlocks} componentsToRender={componentsToRender} />
+);
 
-  return <Blocks blocks={textBlocks} componentsToRender={componentsToRender} />;
-};
+const renderCaption = (block, imageCaptionOffscreenText) => (
+  <Caption>
+    {imageCaptionOffscreenText ? (
+      <VisuallyHiddenText>{imageCaptionOffscreenText}</VisuallyHiddenText>
+    ) : null}
+    {renderText(block)}
+  </Caption>
+);
+
+const renderMultipleCaptions = (blocks, imageCaptionOffscreenText) =>
+  blocks.map(block =>
+    renderCaption(block.model.blocks, imageCaptionOffscreenText),
```

Comment:
Likewise you can use the `deepGet` utility here to safely retrieve the nested blocks. `deepGet(['model', 'blocks'], block)`